miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/ScreenBridgeDispatcher.kt

```diff
@@ -8,37 +8,42 @@ import com.rakuten.tech.mobile.miniapp.js.ErrorBridgeMessage.ERR_SCREEN_ACTION
 
 internal class ScreenBridgeDispatcher(
     private val activity: Activity,
-    private val bridgeExecutor: MiniAppBridgeExecutor
+    private val bridgeExecutor: MiniAppBridgeExecutor,
+    var allowScreenOrientation: Boolean
 ) {
     private var isLocked = false
 
     @Suppress("TooGenericExceptionCaught", "LongMethod")
     fun onScreenRequest(callbackObj: CallbackObj) {
-        try {
-            val requestAction = Gson().fromJson<Screen>(
-                callbackObj.param.toString(),
-                object : TypeToken<Screen>() {}.type
-            ).action
+        if (allowScreenOrientation) {
+            try {
+                val requestAction = Gson().fromJson<Screen>(
+                    callbackObj.param.toString(),
+                    object : TypeToken<Screen>() {}.type
+                ).action
 
-            when (requestAction) {
-                ScreenOrientation.LOCK_LANDSCAPE.value -> {
-                    activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
-                    isLocked = true
+                when (requestAction) {
+                    ScreenOrientation.LOCK_LANDSCAPE.value -> {
+                        activity.requestedOrientation =
+                            ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
+                        isLocked = true
+                    }
+                    ScreenOrientation.LOCK_PORTRAIT.value -> {
+                        activity.requestedOrientation =
+                            ActivityInfo.SCREEN_ORIENTATION_SENSOR_PORTRAIT
+                        isLocked = true
+                    }
+                    ScreenOrientation.LOCK_RELEASE.value -> releaseLock()
                 }
-                ScreenOrientation.LOCK_PORTRAIT.value -> {
-                    activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_PORTRAIT
-                    isLocked = true
-                }
-                ScreenOrientation.LOCK_RELEASE.value -> releaseLock()
+                bridgeExecutor.postValue(callbackObj.id, SUCCESS)
+            } catch (e: Exception) {
+                bridgeExecutor.postError(callbackObj.id, "$ERR_SCREEN_ACTION ${e.message}")
             }
-            bridgeExecutor.postValue(callbackObj.id, SUCCESS)
-        } catch (e: Exception) {
-            bridgeExecutor.postError(callbackObj.id, "$ERR_SCREEN_ACTION ${e.message}")
         }
     }
 
     fun releaseLock() {
-        if (isLocked) {
+        if (allowScreenOrientation && isLocked) {
             activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
```

Comment:
If we use `ActivityInfo.SCREEN_ORIENTATION_USER` here, then doesn't this mean it will override the default for the Host App? I mean if the Host App has set `PORTRAIT`, then this would reset it to `USER` instead. Also, if it's a single activity App, I think this would reset the orientation for the whole App.

Could we instead save the original orientation, and then reset to the original value when `releaseLock()` is called? (Similar to what we were doing in the full-screen view)

Reply:
On fullscreen view code before, I assigned the fixed value for original orientation. Basically, we can only get the portrait or landscape state, we cannot get the screen config state.

There is a situation that Hostapp supports both portrait & landscape but when we reset it to portrait (in case the previous is portrait) -> on android it means we force the portrait lock.
We cannot detect to see whether hostapp config only single or both screen orientations.

Reply:
Could you do something like this?

```
internal class ScreenBridgeDispatcher(
    private val activity: Activity
) {
  private val hostAppOrientation = activity.requestedOrientation

  // ...

  fun releaseLock() {
    activity.requestedOrientation = hostAppOrientation
    // ...
  }
}
```

Reply:
ah yeah it's the right approach 🙇